solutions/CF Mixing Water.mdx

```diff
@@ -0,0 +1,116 @@
+---
+id: cf-1359C
+source: CF
+title: Mixing Water
+author: Jesse Choe
+---
+
+<Info title="Constant Time Solution">
+**Note: the $\mathcal{O}(1)$ solution for this problem is provided in the
+[official editorial](https://codeforces.com/blog/entry/78116).**
+</Info>
+
+Consider the following observations:
+
+- Case 1: pouring an equal number of cups of hot and cold water is equivalent to pouring one cup of each. Otherwise,
+- Case 2: there must be exactly one more cup of hot water poured than cold water poured. Thus, there will be $c + 1$ cups of hot water and $c$ cups of cold water for some $c \geq 0$.
+- The temperature is l+(h-l)(c+1)/(2c+1) when c is odd. This can be proven to be a monotonically decreasing function. Since it's monotonic, we can binary search to approximate the best value for $c$, and test the odd integers aroudn it
```

Comment:
Don't just directly copy it :/

I meant to just phrase it somewhere along these lines.

Reply:
Should I change it?

Reply:
Yes... its very hard to read atm